docs/docs/dagster-basics-tutorial/6-custom-components.md

```diff
@@ -0,0 +1,109 @@
+---
+title: Components
+description: Defining our own components
+sidebar_position: 90
+---
+
+So far we have built out everything via `Definitions`. But this is not the only way to include `Definitions` in a Dagster project.
+
+If we think about the code for our three assets (`customers`, `orders`, and `payments`), it is all very similar. Each asset performs the same action, turning an S3 file into a DuckDB table while differing only in the URL path and table name.
+
+These assets are a great candidate for a [component](/guides/build/components/creating-new-components). Components generate `Definitions` through a configuration layer. There are built-in components that let you integrate with common workflows (such as turning Python scripts into assets) or with popular tools like [dbt](https://www.getdbt.com/) or [Fivetran](https://www.fivetran.com/). There are also custom components that let you define your own specific use cases.
+
+Here we will use a custom component to streamline the development of our assets and replace their `Definitions` in our project with a `Component` which can generate them instead.
```

Comment:
```suggestion
In this step, you will use a custom component to streamline the development of similar assets and replace their `Definitions` in your project with a `Component` that can generate them instead.
```

Reply:
It may be helpful to explain here that the custom component will have a YAML interface -- maybe something like "...and replace their `Definitions` in your project with a `Component` that can generate them from a a YAML configuration file instead"